testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java

```diff
@@ -1117,6 +1117,14 @@ public void searchAndCountGroups() throws Exception {
 
         assertEquals(new Long(allGroups.size()), realm.groups().count(true).get("count"));
         assertEquals(new Long(allGroups.size() + 1), realm.groups().count(false).get("count"));
+        //search and count for group with string group11 -> return 2 top level group, group11 and group0 having subgroup group1111
```

Comment:
@cgeorgilakis  Thank you for the Javadoc and test. It would be also good to cover a scenario which fails without your change. 
```suggestion
        //add another subgroup
        GroupRepresentation level2Group2 = new GroupRepresentation();
        level2Group2.setName("group111111");
        realm.groups().group(firstGroupId).subGroup(level2Group2);

        //search and count for group with string group11 -> return 2 top level group, group11 and group0 having subgroup group1111
        search = realm.groups().groups("group11",0,10);
        assertEquals(2, search.size());
        GroupRepresentation group0 = search.stream().filter(group -> "group0".equals(group.getName())).findAny().orElseGet(null);
        assertNotNull(group0);
        assertEquals(2,group0.getSubGroups().size());
        assertThat(group0.getSubGroups().stream().map(GroupRepresentation::getName).collect(Collectors.toList()), Matchers.containsInAnyOrder("group1111", "group111111"));
        assertEquals(new Long(search.size()), realm.groups().count("group11").get("count"));
```

Reply:
How will we test scenario that fails without my change? 
Fe the corrected test will fail in Keycloak main. realm.groups().count("group11").get("count") will return 3 instead of 2.

Reply:
If you run the test with ConcurrentHashMap group provider `mvn clean install -nsu -B -Pauth-server-undertow -Pmap-storage -f testsuite/integration-arquillian/tests/base/pom.xml -Dtest=GroupTest#searchAndCountGroups` the test will pass if you revert the change in `MapGroupProvider`. The key part is add another subgroup to make it fail without your change.